### PR TITLE
Port twisted.protocols.socks to Python 3

### PR DESCRIPTION
**Pull request by rodrigc**
http://twistedmatrix.com/trac/ticket/8665

### CODE REVIEW

File: twisted/protocols/socks.py [hunk 1]
```diff
@@ -79,35 +85,37 @@ def __init__(self, logging=None, reactor=reactor):
...
         self.buf = self.buf + data
         completeBuffer = self.buf
-        if "\000" in self.buf[8:]:
+        if b"\000" in self.buf[8:]:
             head, self.buf = self.buf[:8], self.buf[8:]
             version, code, port = struct.unpack("!BBH", head[:4])
-            user, self.buf = self.buf.split("\000", 1)
-            if head[4:7] == "\000\000\000" and head[7] != "\000":
+            user, self.buf = self.buf.split(b"\000", 1)
+            if head[4:7] == b"\000\000\000" and head[7] != b"\000":
```
Comment (alex) [inline twisted/protocols/socks.py]: This needs to be `head[7:8]` doesn't it?

Comment (rodrigc) [inline twisted/protocols/socks.py]: @alex you are correct!

Comment (rodrigc) [inline twisted/protocols/socks.py]: @alex I addressed your concern.  Can I merge this?

**Comment by alex** [comment]
👍  lgtm

**Pull request merged**